treeherder/perf/management/commands/reassign_perf_data.py

```diff
@@ -24,19 +29,92 @@ def add_arguments(self, parser):
             '--to',
             action='append',
             help='New signature we want to move performance data to '
-                 '(specify multiple times to get multiple signature)'
+                 '(specify multiple times to get multiple signatures)'
+        )
+        parser.add_argument(
+            '--for',
+            action='store',
+            choices=USE_CASES,
+            metavar='USE CASE',
+            help='''Rename "old" Raptor tp6 subtests, by pointing perf alerts & datum to new signatures.
+                 Cannot be used in conjunction with --from/--to arguments.
+                 Available use cases: {}'''.format(','.join(USE_CASES))
+        )
+        parser.add_argument(
+            '--keep-leftovers',
+            action='store_true',
+            help='Keep database rows even if they become useless after the script runs'
         )
 
     def handle(self, *args, **options):
         from_signatures = options['from']
         to_signatures = options['to']
+        use_case = options['for']
+        keep_leftovers = options['keep_leftovers']
+
+        self.validate_arguments(from_signatures, to_signatures, use_case)
+
+        if use_case:
+            signature_pairs = self.fetch_signature_pairs(use_case)
```

Comment:
what if there is a timeout in fetch_signature_pairs?

Reply:
The script would simply stop its execution (no writing will happen). There's nothing really here we can handle. The dumped database exception will provide enough context to realize that a later 2n re-run could be successful.